Firestore/Swift/Tests/Codable/FirestoreEncoderTests.swift

```diff
@@ -564,26 +564,17 @@ class FirestoreEncoderTests: XCTestCase {
       .failsEncodingWithJSONEncoder()
   }
 
-  func testDecodingDocumentIDWithConfictingFieldsThrows() throws {
+  func testDecodingDocumentIDWithConfictingFieldsDoesNotThrow() throws {
     struct Model: Codable, Equatable {
       var name: String
       @DocumentID var docId: DocumentReference?
     }
 
-    do {
-      _ = try Firestore.Decoder().decode(
-        Model.self,
-        from: ["name": "abc", "docId": "Causing conflict"],
-        in: FSTTestDocRef("abc/123")
-      )
-      XCTFail("Failed to throw")
-    } catch let FirestoreDecodingError.fieldNameConflict(msg) {
-      XCTAssertEqual(msg, "Field name [\"docId\"] was found from document \"abc/123\", "
-        + "cannot assign the document reference to this field.")
-      return
-    } catch {
-      XCTFail("Unrecognized error: \(error)")
-    }
+    _ = try Firestore.Decoder().decode(
```

Comment:
Should this be considered a breaking change?

Reply:
I'd say 'no', but I can see how this could be contended.
If you rely on decoding to fail, then this is technically breaking, but I cannot see how that would be meaningful.

On the other hand I guess there would be no harm in documenting it as a breaking change and explaining the benefits for not failing in this case (easier to model backwards and forwards compatibility).

Reply:
At first I was nervous of this change in functionality because throwing an error alerts me to a programming error (a conflicting field name) but on thinking through backwards compatibility, I'm leaning towards this being a big improvement.

**Scenario**
We have a user with a name and a location (shown below). In the future, someone decides to add a user ID to the model and adds a field `id` in the Firestore database.
```
struct User: Codable, Identifiable {
  @DocumentID var id: String?
  var name: String
  var location: String?
}
```

Previous Approach:
- Existing clients would crash when the conflicting field name is found.

New Approach:
- Existing clients continue functioning like normal -- they're unlikely to have a UI for displaying user IDs since the field didn't exist when they were created.
- When the developer adds user IDs to their in-app model, they'd realize the field named `id` already exists in their `Codable` and rename it.

I don't know whether to consider it a breaking change or not either. I don't feel strongly either way, as long as it ends up in the changelog regardless. Potentially there are other scenarios I'm not thinking of where throwing would be beneficial though.

Reply:
@andrewheard you're exactly right. With the crashing behavior any migration to using document ids or away from using document ids is not possible - and you'd have to invent a new property name to avoid clashes and crashes.

So if you wanted to migrate in the other direction: from using an explicit id to using a `@DocumentId`, you'd have a similar, but opposite situation of new clients crashing on old data.

For these reasons I don't think that the crashing behavior belongs in the realm of 'programmer errors' as model evolution is a quite natural part of the evolution of a program - and many other types of model evolutions can be handled gracefully.